mip/mip.c

```diff
@@ -180,7 +180,7 @@ static void q_copyout(struct queue *q, uint8_t *buf, size_t len, size_t tail) {
 
 static bool q_write(struct queue *q, const void *buf, size_t len) {
   bool success = false;
-  size_t left = q->len - q->head + q->tail;
+  size_t left = (q->len - q->head + q->tail -1) % q->len;
```

Comment:
Can you add a unit test to tests/mip_test.c , that tests queue, that fails without this change , and passes with this change, please?

Reply:
Done, please see #1719

Reply:
Line 26 passes with this patch applied